### PR TITLE
rename ort to maia in dynamo's ort backend.

### LINKED ISSUE #124966: torch dynamo's ORT backend uses "ort" dispatch key instead of "maia"
### 🐛 Describe the bug

https://github.com/pytorch/pytorch/blob/b8b04b26fbf160874f7f1a9db61e49801fd4fcbe/torch/onnx/_internal/onnxruntime.py#L148

this should have been fixed as part of https://github.com/pytorch/pytorch/pull/123265, but it was missed.

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
Clang version: Could not collect
CMake version: version 3.27.0
Libc version: glibc-2.31

Python version: 3.8.10 (default, Nov 22 2023, 10:22:35)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-101-generic-x86_64-with-glibc2.29
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      46 bits physical, 48 bits virtual
CPU(s):                             12
On-line CPU(s) list:                0-11
Thread(s) per core:                 2
Core(s) per socket:                 6
Socket(s):                          1
NUMA node(s):                       1
Vendor ID:                          GenuineIntel
CPU family:                         6
Model:                              85
Model name:                         Intel(R) Xeon(R) W-2235 CPU @ 3.80GHz
Stepping:                           7
CPU MHz:                            3800.000
CPU max MHz:                        4600.0000
CPU min MHz:                        1200.0000
BogoMIPS:                           7599.80
Virtualization:                     VT-x
L1d cache:                          192 KiB
L1i cache:                          192 KiB
L2 cache:                           6 MiB
L3 cache:                           8.3 MiB
NUMA node0 CPU(s):                  0-11
Vulnerability Gather data sampling: Mitigation; Microcode
Vulnerability Itlb multihit:        KVM: Mitigation: VMX disabled
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Retbleed:             Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Mitigation; TSX disabled
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cdp_l3 invpcid_single intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm mpx rdt_a avx512f avx512dq rdseed adx smap clflushopt clwb intel_pt avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512_vnni md_clear flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] flake8==7.0.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.4
[pip3] onnx==1.16.0
[pip3] onnxruntime-training==0.20240418.1+cpu
[pip3] onnxscript==0.1.0.dev20240328
[pip3] torch==2.4.0.dev20240423+cpu
[pip3] torchvision==0.19.0.dev20240423+cpu
[pip3] triton==3.0.0
[conda] No relevant packages

### PR DESCRIPTION
**Pull request by ashari4**
Fixes #124966

### CODE REVIEW

**Review by thiagocrepaldi — APPROVED** [review]

File: test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py [hunk 1]
```diff
@@ -50,6 +50,11 @@ def tearDown(self):
         torch._dynamo.reset()
         OrtBackend.clear_cached_instances()
 
+    def test_get_ort_device_type(self):
+        self.assertEqual(onnxruntime._get_ort_device_type("cuda"), onnxruntime.ORTC.OrtDevice.cuda())
+        self.assertEqual(onnxruntime._get_ort_device_type("cpu"), onnxruntime.ORTC.OrtDevice.cpu())
+        self.assertEqual(onnxruntime._get_ort_device_type("maia"), onnxruntime.ORTC.OrtDevice.npu())
+
```
Comment (thiagocrepaldi) [inline test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py:57]:
```suggestion

```

File: test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py [hunk 2]
```diff
@@ -50,6 +50,10 @@ def tearDown(self):
         torch._dynamo.reset()
         OrtBackend.clear_cached_instances()
 
+    def test_get_ort_device_type(self):
+        self.assertEqual(onnxruntime._get_ort_device_type("cuda"), onnxruntime.ORTC.OrtDevice.cuda())
+        self.assertEqual(onnxruntime._get_ort_device_type("cpu"), onnxruntime.ORTC.OrtDevice.cpu())
+        self.assertEqual(onnxruntime._get_ort_device_type("maia"), onnxruntime.ORTC.OrtDevice.npu())
```
Comment (thiagocrepaldi) [inline test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py:56]:
```suggestion
        self.assertEqual(onnxruntime._get_ort_device_type("maia"), onnxruntime.ORTC.OrtDevice.npu())

```

**Comment by thiagocrepaldi** [comment]
@ashari4 can you

```bash
cd /path/to/pytorch/repor
git pull
lintrunner init
lintrunner -a
git push
```

The CI is broken because of code style and we need to fix before merging it

**Comment by ashari4** [comment]
@thiagocrepaldi could i please get your help to reinitiate the merge again?

**Pull request closed without merging**